clients/drcachesim/tests/schedule_stats_test.cpp

```diff
@@ -324,12 +341,92 @@ test_cpu_footprint()
     return true;
 }
 
+static bool
+test_syscall_latencies()
+{
+    static constexpr int64_t TID_A = 42;
+    static constexpr int64_t TID_B = 142;
+    static constexpr int64_t TID_C = 242;
+    static constexpr uintptr_t SYSNUM_X = 12;
+    static constexpr uintptr_t SYSNUM_Y = 167;
+    std::vector<std::vector<memref_t>> memrefs = {
+        {
+            gen_instr(TID_A),
+            // Involuntary switch.
+            gen_instr(TID_B),
+            gen_marker(TID_B, TRACE_MARKER_TYPE_TIMESTAMP, 1100),
+            gen_marker(TID_B, TRACE_MARKER_TYPE_SYSCALL, SYSNUM_X),
+            gen_marker(TID_B, TRACE_MARKER_TYPE_TIMESTAMP, 1600),
+            // Voluntary switch, on non-maybe-blocking-marked syscall.
+            gen_instr(TID_A),
+            gen_instr(TID_A),
+            gen_instr(TID_A),
+            gen_marker(TID_A, TRACE_MARKER_TYPE_TIMESTAMP, 2100),
+            gen_marker(TID_A, TRACE_MARKER_TYPE_SYSCALL, SYSNUM_Y),
+            gen_marker(TID_A, TRACE_MARKER_TYPE_MAYBE_BLOCKING_SYSCALL, 0),
+            gen_marker(TID_A, TRACE_MARKER_TYPE_DIRECT_THREAD_SWITCH, TID_C),
+            gen_marker(TID_A, TRACE_MARKER_TYPE_TIMESTAMP, 2300), // Direct switch.
+            gen_instr(TID_C),
+            // No switch: latency too small.
```

Comment:
nit: Move this "No switch ..." comment to after the timestamp-2599 marker (because that's where a switch would have been if the latency were high enough)

Reply:
Done.